packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -30,12 +30,18 @@ class BlockListBlock extends Component {
 
 		this.insertBlocksAfter = this.insertBlocksAfter.bind( this );
 		this.onFocus = this.onFocus.bind( this );
+		this.onSettingsToggle = this.onSettingsToggle.bind( this );
 
 		this.state = {
 			isFullyBordered: false,
+			showSettings: false,
 		};
 	}
 
+	onSettingsToggle() {
+		this.setState( { showSettings: ! this.state.showSettings } );
+	}
+
```

Comment:
Move `showSettings` and settingsToggle from edit.native.js of image blok to allow open the settings from each block passing onSettingsToggle.

Currently I'm working to move this logic to `InspectorControls` context provider so perhaps it will be possible to do not change `block.native.js`

Reply:
The thing that bugs me is that `showSettings` flag is not used for anything inside this component, it is just passed to `BlockMobileToolbar`. What is more it is toggled by a component outside of this component. So it looks like is doesn't belong here, right?

My [idea](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1300#issuecomment-527127376) was to create a new SettingsToolbarButton component that renders below button and keep `showSettings` in its state but I don't have strong opinions about that.

```
                                               <ToolbarButton
							title={ __( 'Image Settings' ) }
							icon="admin-generic"
							onClick={ onSettingsToggle }
						/>
```